dashboard/src/components/LoginForm/LoginForm.v2.tsx

```diff
@@ -0,0 +1,145 @@
+import { ClarityIcons, infoCircleIcon } from "@clr/core/icon-shapes";
+import { Location } from "history";
+import * as React from "react";
+import { Redirect } from "react-router";
+import { CdsButton } from "../Clarity/clarity";
+import { CdsIcon } from "../Clarity/clarity";
+
+import LoadingWrapper from "../../components/LoadingWrapper";
+import "./LoginForm.v2.css";
+
+ClarityIcons.addIcons(infoCircleIcon);
+
+interface ILoginFormProps {
+  authenticated: boolean;
+  authenticating: boolean;
+  authenticationError: string | undefined;
+  oauthLoginURI: string;
+  authenticate: (token: string) => any;
+  checkCookieAuthentication: () => void;
+  location: Location;
+}
+
+interface ILoginFormState {
+  token: string;
+}
+
+class LoginForm extends React.Component<ILoginFormProps, ILoginFormState> {
+  public state: ILoginFormState = { token: "" };
+
+  public componentDidMount() {
+    if (this.props.oauthLoginURI) {
+      this.props.checkCookieAuthentication();
+    }
+  }
+
+  public render() {
+    if (this.props.authenticating) {
+      return <LoadingWrapper />;
+    }
+    if (this.props.authenticated) {
+      const { from } = (this.props.location.state as any) || { from: { pathname: "/" } };
+      return <Redirect to={from} />;
+    }
+
+    return (
+      <div className="login-wrapper">
+        <form className="login clr-form" onSubmit={this.handleSubmit}>
+          {this.props.oauthLoginURI ? this.oauthLogin() : this.tokenLogin()}
+          <div className="login-moreinfo">
+            <a
+              href="https://github.com/kubeapps/kubeapps/blob/master/docs/user/access-control.md"
+              target="_blank"
+              rel="noopener noreferrer"
+            >
+              <CdsIcon shape="info-circle" />
+              More Info
+            </a>
+          </div>
+        </form>
+      </div>
+    );
+  }
+
+  private handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const { token } = this.state;
+    return token && (await this.props.authenticate(token));
+  };
+
+  private handleTokenChange = (e: React.FormEvent<HTMLInputElement>) => {
+    this.setState({ token: e.currentTarget.value });
+  };
+
+  private oauthLogin = () => {
+    return (
+      <>
+        <section className="title">
+          <h3 className="welcome">Welcome to</h3>
+          Kubeapps
+          <h5 className="hint">
+            Your cluster operator has enabled login via an authentication provider.
+          </h5>
+        </section>
```

Comment:
`section` blocks should be used to identify a block of a website that acts as a single unit. In this case, the "title" doesn't represent a block, but the full login block does.

Please, update the `section` to wrap all this block instead only the title. 

In addition to that, ensure you create a title that represents the block itself. For a screen reader, the current heading title (`Welcome to`) doesn't provide enough information. `Kubeapps` string should be part of this title too. You can use a `span` tag to change the style of the `Kubeapps` string if it's required. 

Regarding the hint, a heading is not an appropriate HTML tag for it. You can use a paragraph for this.

Once the previous comments are done, you should link the section with both title and hint sections. For this, give an ID to them and link them using the aria properties:

```
<section aria-labelledby="login-title" aria-describedby="login-desc">
  <h3 id="login-title">
    Welcome to
    <span>Kubeapps</span>
  </h3>
  <p id="login-desc">
    ...
  </p>
  <div>
    ... Login form ...
  </div>
</section>
```

Reply:
There are still some work about accessibility in the Clarity docs. The approach I suggested is more appropriate for this case.